includes/Performance.php

```diff
@@ -134,19 +134,24 @@ public function hooks() {
 			add_filter( 'got_rewrite', '__return_true' );
 		}
 
-		add_filter(
-			'mod_rewrite_rules',
-			function ( $content ) {
-				add_action(
-					'shutdown',
-					function () {
-						do_action( 'newfold_update_htaccess' );
-					}
-				);
-
-				return $content;
-			}
-		);
+		if (
+			isset( $_REQUEST['action'], $_REQUEST['plugin'] ) &&
+            'activate' === $_REQUEST['action'] &&
```

Comment:
@AleTorrisi, based on the PR description, should this run on deactivation and not `activate`? And should we also hook into something like `deactivated_plugin` to handle cases where the plugin is deactivated via WP-CLI? Let me know. Thanks!